src/ZeroLog/LogEvent.cs

```diff
@@ -16,19 +16,24 @@ internal unsafe partial class LogEvent : IInternalLogEvent
         private const int _maxArgCapacity = byte.MaxValue;
         private string[] _strings;
         private IntPtr[] _argPointers;
+        private IntPtr[] _keyValuePointers;
```

Comment:
You don't need one of those per log event - that's expensive memory-wise. Change the signature of `WriteToStringBuffer`, so you can have a single array or list instance.

Reply:
That seems much better, I've updated this, please take a look :)